runtime/src/main/java/org/corfudb/util/MetricsUtils.java

```diff
@@ -199,17 +239,38 @@ public static void addCacheGauges(@NonNull MetricRegistry metrics,
         }
     }
 
-    public static void addJvmMetrics(@NonNull MetricRegistry metrics, String pfx) {
+    // If enabled, setup reporting of JVM metrics including garbage collection,
+    // memory, and thread statistics.
+    private static void setupJvmMetrics(@NonNull MetricRegistry metrics) {
+        if (!metricsJvmCollectionEnabled) return;
+
         try {
-            metrics.register(pfx + "jvm.gc", metricsJVMGC);
-            metrics.register(pfx + "jvm.memory", metricsJVMMem);
-            metrics.register(pfx + "jvm.thread", metricsJVMThread);
-            metrics.register(pfx + "jvm.file-descriptors-used", metricsJVMFdGauge);
+            metrics.register("jvm.gc", metricsJVMGC);
+            metrics.register("jvm.memory", metricsJVMMem);
+            metrics.register("jvm.thread", metricsJVMThread);
+            metrics.register("jvm.file-descriptors-used", metricsJVMFdGauge);
         } catch (IllegalArgumentException e) {
             // Re-registering metrics during test runs, not a problem
         }
     }
 
+    // If enabled, collect reports on logger usage statistics.
+    private static void setupLogAnalysisMetrics(@NonNull MetricRegistry metrics) {
```

Comment:
was this a concern. Why do we need metrics for logger's performance ?

Reply:
@medhavidhawan @Maithem Log analysis provides an aggregate of log messages sent to the main logger. It doesn’t provide a fine grain information on the status of errors or warning. However it trivial to monitor and locate existence or spikes of errors or warnings being logged. This might be a useful troubleshooting tool but if you are not convinced of the value I will remove it.